solarforecastarbiter/io/api.py

```diff
@@ -775,13 +775,15 @@ def process_report_dict(self, rep_dict):
         for o in rep_params['object_pairs']:
             fx = self.get_forecast(o['forecast'])
             norm = o.get('normalization')
+            unc = o.get('uncertainty')
             if 'observation' in o:
                 obs = self.get_observation(o['observation'])
                 pair = datamodel.ForecastObservation(
-                    fx, obs, normalization=norm)
+                    fx, obs, normalization=norm, uncertainty=unc)
             elif 'aggregate' in o:
                 agg = self.get_aggregate(o['aggregate'])
-                pair = datamodel.ForecastAggregate(fx, agg, normalization=norm)
+                pair = datamodel.ForecastAggregate(
+                    fx, agg, normalization=norm, uncertainty=unc)
```

Comment:
@lboeman since we're not using `from_dict`, I need to handle None/float/str types here, right? I think this needs to be done for `normalization` too.

Reply:
Yeah, You should only get a string or None back from the API.  So you'd have to handle the cast to float here or in `__set_uncertainty__` I feel like the latter might make more sense since we're already checking if the field is a string and maybe we should check for the [0, 100] bounds too?

Reply:
Thanks. For normalization it turns out I'm already doing this in `__set_normalization__` 🤦‍♂  my brain doesn't work anymore.